src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -0,0 +1,484 @@
+		return this.rightFootAcceleration.length();
+	}
+
+	public float getLeftFootAccelerationY() {
+		return this.leftFootAcceleration.y;
+	}
+
+	public float getRightFootAccelerationY() {
+		return this.rightFootAcceleration.y;
+	}
+
+	public void setDetectionMode(int mode) {
+		this.detectionMode = mode;
+	}
+
+	// calculate momvent attributes
+	public void calculateFootAttributes(boolean active) {
+		updateFrameNumber(0);
+
+		// compute attributes of the legs
+		computeVelocity();
+		computeAccelerationMagnitude();
+
+		// check if the acceleration triggers forced unlock
+		if (detectionMode == FOOT_ACCEL) {
+			computeAccelerationAboveThresholdFootTrackers();
+		} else {
+			computeAccelerationAboveThresholdAnkleTrackers();
+		}
+
+		// calculate the scalar for other parameters
+		leftFootSensitivity = getLeftFootScalar();
+		rightFootSensitivity = getRightFootScalar();
+
+		// if correction is inactive state is unknown (default to unlocked)
+		if (!active) {
+			leftLegState = UNLOCKED;
+			rightLegState = UNLOCKED;
+		} else {
+			computeState();
+		}
+	}
+
+	// update the frame number of all the frames
+	public void updateFrameNumber(int frameNumber) {
+		this.frameNumber = frameNumber;
+		if (this.frameNumber >= 10) {
+			this.parent = null; // once a frame is 10 frames old, it is no
+								// longer
+								// needed
+		}
+		if (parent != null) {
+			parent.updateFrameNumber(frameNumber + 1);
+		}
+	}
+
+	// compute the state of the legs
+	private void computeState() {
+		// based on the last state of the legs compute their state for this
+		// individual frame
+		leftLegState = checkStateLeft();
+
+		rightLegState = checkStateRight();
+	}
+
+	// check if a locked foot should stay locked or be released
+	private int checkStateLeft() {
+		float timeStep = 1.0f / ((timeOfFrame - parent.timeOfFrame) / 1000000000.0f);
+		if (parent.leftLegState == UNLOCKED) {
+			if (
+				parent.getLeftFootHorizantalDifference() > SKATING_CUTOFF_ENGAGE
+					|| leftFootVelocityMagnitude * timeStep > SKATING_VELOCITY_CUTOFF_ENGAGE
+					|| leftFootAngleDiff * timeStep > SKATING_ROTATIONAL_VELOCITY_CUTOFF_ENGAGE
+					|| leftFootPosition.y > leftFloorLevel + FLOOR_DISTANCE_CUTOFF
+					|| accelerationAboveThresholdLeft
+			) {
+				return UNLOCKED;
+			}
+			return LOCKED;
+
+		} else {
+			if (
+				parent.getLeftFootHorizantalDifference() > SKATING_DISTANCE_CUTOFF
+					|| leftFootVelocityMagnitude * timeStep
+						> SKATING_VELOCITY_THRESHOLD * leftFootSensitivity
+					|| leftFootAngleDiff * timeStep
+						> SKATING_ROTVELOCITY_THRESHOLD * leftFootSensitivity
+					|| leftFootPosition.y > leftFloorLevel + FLOOR_DISTANCE_CUTOFF
+					|| accelerationAboveThresholdLeft
+			) {
+				return UNLOCKED;
+			}
+			return LOCKED;
+		}
+	}
+
+	// check if a locked foot should stay locked or be released
+	private int checkStateRight() {
+		float timeStep = 1.0f / ((timeOfFrame - parent.timeOfFrame) / 1000000000.0f);
```

Comment:
See the comment above "Can we have a comment or constant explaining `1000000000.0f`?".